casanovo/casanovo.py

```diff
@@ -179,5 +207,105 @@ def main(
         model_runner.train(peak_path, peak_path_val, model, config)
 
 
+def _get_model_weights() -> str:
+    """
+    Use cached model weights or download them from GitHub.
+
+    If no weights file (extension: .ckpt) is available in the cache directory,
+    it will be downloaded from a release asset on GitHub.
+    Model weights are retrieved by matching release version. If no model weights
+    for an identical release (major, minor, patch), alternative releases with
+    matching (i) major and minor, or (ii) major versions will be used.
+    If no matching release can be found, no model weights will be downloaded.
+
+    Note that the GitHub API is limited to 60 requests from the same IP per
+    hour.
+
+    Returns
+    -------
+    str
+        The name of the model weights file.
+    """
+    cache_dir = appdirs.user_cache_dir("casanovo", False, opinion=False)
+    os.makedirs(cache_dir, exist_ok=True)
+    version = tuple(
+        g
+        for g in re.match(
+            r"(\d+)\.(\d+)\.(\d+)(?:.dev\d+.+)?", __version__
+        ).groups()
+    )
```

Comment:
**suggestion:** This is pretty ugly. Maybe use `re.compile` above to clean it up? Something like:
```python
version_regex = re.compile(r"(\d+)\.(\d+)\.(\d+)(?:.dev\d+.+)?")
version = tuple(g for g in version_regex.match(__version__).groups())
```